crates/sui-core/src/authority_aggregator.rs

```diff
@@ -858,6 +948,167 @@ where
+                        };
+
+                        // Return all errors if a quorum is not possible.
+                        if state.bad_weight > validity {
+                            return Err(SuiError::TooManyIncorrectAuthorities {
+                                errors: state.errors,
+                                action: "get_committee_info",
+                            });
+                        }
+                        Ok(ReduceOutput::Continue(state))
+                    })
+                },
+                // A long timeout before we hear back from a quorum
+                self.timeouts.pre_quorum_timeout,
+            )
+            .await?;
+
+        if let Some(committee_info) = final_state.committee_info {
+            Ok(committee_info)
+        } else {
+            Err(SuiError::TooManyIncorrectAuthorities {
+                errors: final_state.errors,
+                action: "get_committee_info",
+            })
+        }
+    }
+
+    /// Query validators for latest SuiSystemState and try to form
+    /// CommitteeWithNetworkAddress. Only return Some(CommitteeWithNetAddresses)
+    /// when there is quorum. This function tolerates uninteresting
+    /// differences in SuiSystemState as long as they all link to the same
+    /// CommitteeWithNetAddresses.
+    /// This function ignores SuiSystemState that has older epoch than
+    /// `minimal_epoch`.
+    /// Usually, the caller should pass in the local incumbent epoch id
+    /// as `minimal_epoch`, to avoid getting confused by byzantine
+    /// validators.
+    pub async fn get_committee_with_net_addresses(
+        &self,
+        minimal_epoch: EpochId,
+    ) -> SuiResult<CommitteeWithNetAddresses> {
+        let (aggregate_object_info, _certificates) =
+            // Skip committee check because this call usually happens when there's a potential new epoch
+            self.get_object_by_id(SUI_SYSTEM_STATE_OBJECT_ID, true).await?;
+
+        let mut committee_and_sigs = aggregate_object_info
+            .into_iter()
+            .filter_map(
+                |(
+                    (_object_ref, _transaction_digest),
+                    (object_option, _layout_option, object_authorities),
+                )| {
+                    if let Some(object) = object_option {
+                        let system_state = object.data.try_as_move().and_then(|move_object| {
+                            bcs::from_bytes::<SuiSystemState>(move_object.contents()).ok()
+                        })?;
+                        if system_state.epoch < minimal_epoch {
+                            None
+                        } else {
+                            let committee = system_state.get_current_epoch_committee();
+                            Some((committee, object_authorities))
+                        }
+                    } else {
+                        None
+                    }
+                },
+            )
+            .collect::<Vec<_>>();
```

Comment:
shouldn't this be grouping the committee info by epoch? I assume from the if stmt on line 1060 that different validators can return the state from different epochs (e.g. if the txn updating the system object hasn't executed on all validators). Couldn't this then group votes from different epochs together?